# HG changeset patch
# User fyang
# Date 1504609758 -28800
#      Tue Sep 05 19:09:18 2017 +0800
# Node ID e2d13b496321f87a9232573d5671af86311af99f
# Parent  bed72f56b47da796c83de7e873ce0a9ac44b0af6
8187224, PR3488: aarch64: some inconsistency between aarch64_ad.m4 and aarch64.ad
Summary: fix ROL_INSN and ROR_INSN definition in aarch64_ad.m4
Reviewed-by: aph

diff --git a/src/cpu/aarch64/vm/aarch64_ad.m4 b/src/cpu/aarch64/vm/aarch64_ad.m4
--- openjdk/hotspot/src/cpu/aarch64/vm/aarch64_ad.m4
+++ openjdk/hotspot/src/cpu/aarch64/vm/aarch64_ad.m4
@@ -268,21 +268,21 @@
   ins_pipe(ialu_reg_reg_vshift);
 %}')dnl
 define(ROL_INSN, `
-instruct $3$1_rReg_Var_C$2(iRegLNoSp dst, iRegL src, iRegI shift, immI$2 c$2, rFlagsReg cr)
+instruct $3$1_rReg_Var_C$2(iReg$1NoSp dst, iReg$1 src, iRegI shift, immI$2 c$2, rFlagsReg cr)
 %{
   match(Set dst (Or$1 (LShift$1 src shift) (URShift$1 src (SubI c$2 shift))));
 
   expand %{
-    $3L_rReg(dst, src, shift, cr);
+    $3$1_rReg(dst, src, shift, cr);
   %}
 %}')dnl
 define(ROR_INSN, `
-instruct $3$1_rReg_Var_C$2(iRegLNoSp dst, iRegL src, iRegI shift, immI$2 c$2, rFlagsReg cr)
+instruct $3$1_rReg_Var_C$2(iReg$1NoSp dst, iReg$1 src, iRegI shift, immI$2 c$2, rFlagsReg cr)
 %{
   match(Set dst (Or$1 (URShift$1 src shift) (LShift$1 src (SubI c$2 shift))));
 
   expand %{
-    $3L_rReg(dst, src, shift, cr);
+    $3$1_rReg(dst, src, shift, cr);
   %}
 %}')dnl
 ROL_EXPAND(L, rol, rorv)
